packages/server/lib/controllers/auth/postBill.ts

```diff
@@ -0,0 +1,209 @@
+import type { NextFunction } from 'express';
+import { z } from 'zod';
+import { asyncWrapper } from '../../utils/asyncWrapper.js';
+import { zodErrorToHTTP, stringifyError } from '@nangohq/utils';
+import {
+    analytics,
+    configService,
+    AnalyticsTypes,
+    getConnectionConfig,
+    connectionService,
+    errorManager,
+    ErrorSourceEnum,
+    LogActionEnum,
+    getProvider
+} from '@nangohq/shared';
+import type { PostPublicBillAuthorization } from '@nangohq/types';
+import type { LogContext } from '@nangohq/logs';
+import { defaultOperationExpiration, logContextGetter } from '@nangohq/logs';
+import { hmacCheck } from '../../utils/hmac.js';
+import { connectionCreated as connectionCreatedHook, connectionCreationFailed as connectionCreationFailedHook } from '../../hooks/hooks.js';
+import { connectSessionTokenSchema, connectionIdSchema, providerConfigKeySchema } from '../../helpers/validation.js';
+
+const bodyValidation = z
+    .object({
+        username: z.string().min(1),
+        password: z.string().min(1),
+        organization_id: z.string().min(1),
+        dev_key: z.string().min(1)
+    })
+    .strict();
+
+const queryStringValidation = z
+    .object({
+        connection_id: connectionIdSchema.optional(),
+        params: z.record(z.any()).optional(),
+        public_key: z.string().uuid().optional(),
+        connect_session_token: connectSessionTokenSchema.optional(),
+        user_scope: z.string().optional(),
+        hmac: z.string().optional()
+    })
+    .strict();
+
+const paramValidation = z
+    .object({
+        providerConfigKey: providerConfigKeySchema
+    })
+    .strict();
+
+export const postPublicBillAuthorization = asyncWrapper<PostPublicBillAuthorization>(async (req, res, next: NextFunction) => {
+    const val = bodyValidation.safeParse(req.body);
+    if (!val.success) {
+        res.status(400).send({
+            error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) }
+        });
+        return;
+    }
+
+    const queryStringVal = queryStringValidation.safeParse(req.query);
+    if (!queryStringVal.success) {
+        res.status(400).send({
+            error: { code: 'invalid_query_params', errors: zodErrorToHTTP(queryStringVal.error) }
+        });
+        return;
+    }
+
+    const paramVal = paramValidation.safeParse(req.params);
```

Comment:
`paramsVal` and `paramsValidation` with a `s`, no?

Reply:
Yes it should be `paramsVal` and `paramsValidation` , updated this.